ironic/resource_ironic_node_v1.go

```diff
@@ -348,6 +354,22 @@ func resourceNodeV1Update(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if d.HasChange("properties") || d.HasChange("root_device") {
+		properties := d.Get("properties").(map[string]interface{})
+		rootDisk := d.Get("root_device").(map[string]interface{})
+		properties["root_device"] = rootDisk
```

Comment:
This is used multiple times, could we extract it to a helper function? It would make it easier later if we have additional properties that need this same treatment

Reply:
Sure will do!